src/probability_theory/integration.lean

```diff
@@ -27,38 +27,38 @@ namespace probability_theory
    `E[f * indicator T c 0]=E[f] * E[indicator T c 0]`. It is useful for
    `lintegral_mul_eq_lintegral_mul_lintegral_of_independent_measurable_space`. -/
 lemma lintegral_mul_indicator_eq_lintegral_mul_lintegral_indicator
-  {Mf : measurable_space α} [M : measurable_space α] {μ : measure α}
-  (hMf : Mf ≤ M) (c : ℝ≥0∞) {T : set α} (h_meas_T : M.measurable_set' T)
+  {Mf : measurable_space α} [M : measurable_space α] {μ : measure α} (hMf : Mf ≤ M)
+  (c : ℝ≥0∞) {T : set α} (h_meas_T : measurable_set T)
```

Comment:
Are you sure that lean chooses the right `measurable_space` instance for `measurable_set` here? I think it's probably best to stick to the explicit primed version rather than relying on the reader knowing how duplicate typeclasses are resolved.

Reply:
There is the same issue with the measure, which is also defined with respect to the last measurable_space.
I can use `@measurable_set _ M T` to make it explicit. I think that is preferable to `M.measurable_set' T`.

Reply:
Maybe the thing to do is just explain in the module docstring something like

> Many lemmas in this file take two arguments of the same typeclass. It is worth remembering that lean will always pick the later typeclass in this situation, and does not care whether the arguments are `[]`, `{}`, or `()`: All of these use the measure `M2` to define `μ`:
>
> ```
> {M1 : measurable_space α} [M2 : measurable_space α] {μ : measure α}
> [M1 : measurable_space α] {M2 : measurable_space α} {μ : measure α}
> ```